src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.h

```diff
@@ -0,0 +1,255 @@
+ */
+class TlsCertificateManagementDelegate
+{
+public:
+    using RootCertStructType             = TlsCertificateManagement::Structs::TLSCertStruct::DecodableType;
+    using ClientCertStructType           = TlsCertificateManagement::Structs::TLSClientCertificateDetailStruct::DecodableType;
+    using ProvisionRootCertificateType   = TlsCertificateManagement::Commands::ProvisionRootCertificate::DecodableType;
+    using ProvisionClientCertificateType = TlsCertificateManagement::Commands::ProvisionClientCertificate::DecodableType;
+    using ClientCsrType                  = TlsCertificateManagement::Commands::TLSClientCSR::DecodableType;
+    using ClientCsrResponseType          = TlsCertificateManagement::Commands::TLSClientCSRResponse::Type;
+
+    TlsCertificateManagementDelegate() = default;
+
+    virtual ~TlsCertificateManagementDelegate() = default;
+
+    using LoadedRootCertificateCallback   = std::function<CHIP_ERROR(Tls::CertificateTable::RootCertStruct & cert)>;
+    using LoadedClientCertificateCallback = std::function<CHIP_ERROR(Tls::CertificateTable::ClientCertStruct & cert)>;
+    using GeneratedCsrCallback            = std::function<Protocols::InteractionModel::Status(ClientCsrResponseType & cert)>;
+
+    /**
+     * @brief Appends a root certificate to the provisioned root certificates list maintained by the delegate.
+     *        The delegate must ensure it makes a copy of the provided request and data, if any.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] provisionReq The request data specifying the root certificate to be provisioned
+     * @param[out] outCaid a reference to the TLSCAID variable that is to contain the ID of the provisioned root cert.
+     *
+     * @return CHIP_NO_ERROR if the certificate was appended to the list successfully.
+     * @return CHIP_ERROR if there was an error adding the certificate.
+     */
+    virtual Protocols::InteractionModel::ClusterStatusCode ProvisionRootCert(EndpointId matterEndpoint, FabricIndex fabric,
+                                                                             const ProvisionRootCertificateType & provisionReq,
+                                                                             Tls::TLSCAID & outCaid) = 0;
+
+    /**
+     * @brief Executes loadedCallback for each root certificate. The certificate passed to loadedCallback has a guaranteed lifetime
+     * of the method call.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] loadedCallback lambda to execute for each certificate.  If this function returns an error result,
+     * iteration stops and returns that same error result.
+     */
+    virtual CHIP_ERROR LoadedRootCerts(EndpointId matterEndpoint, FabricIndex fabric,
```

Comment:
I don't quite understand why this is "Loaded" in the past tense.  It sounds like a notification, but that's not what this is, right?

Reply:
Somewhat - basically it loads the cert into memory from persistence, thus it is a callback/notification of "the cert has been loaded from persistence -> memory"

But I can change it to something less weird, maybe `ForEachRootCert`?

Reply:
Tracked @ https://github.com/project-chip/connectedhomeip/issues/40362